src/api/spec/controllers/staging/staged_requests_controller_spec.rb

```diff
@@ -99,6 +99,65 @@
       it { expect(staging_project.staged_requests).to include(bs_request) }
       it { assert_select 'status[code=ok]' }
     end
+
+    context 'with an excluded request', vcr: true do
+      context 'when not providing the remove exclusion parameter' do
+        before do
+          create(:request_exclusion,
+                 bs_request: bs_request,
+                 number: bs_request.number,
+                 staging_workflow: staging_workflow)
+          login user
+          post :create,
+               params: { staging_workflow_project: staging_workflow.project.name,
+                         staging_project_name: staging_project.name, format: :xml },
+               body: "<requests><number>#{bs_request.number}</number></requests>"
+        end
+
+        it { expect(response).to have_http_status(:bad_request) }
+        it { expect(response.body).to match(/invalid_request/) }
+        it { expect(response.body).to match(/Error while removing excluded requests: Request #{bs_request.number} not found in Staging for project #{staging_workflow.project.name}/) }
+      end
+
+      context 'when providing the remove exclusion parameter' do
+        before do
+          create(:request_exclusion,
+                 bs_request: bs_request,
+                 number: bs_request.number,
+                 staging_workflow: staging_workflow)
+          login user
+          post :create,
+               params: { staging_workflow_project: staging_workflow.project.name,
+                         staging_project_name: staging_project.name,
+                         remove_exclusion: true,
+                         format: :xml },
+               body: "<requests><number>#{bs_request.number}</number></requests>"
+        end
+
+        it { expect(response).to have_http_status(:success) }
+        it { expect(staging_project.packages.pluck(:name)).to match_array([target_package.name]) }
+        it { expect(staging_project.staged_requests).to include(bs_request) }
+        it { assert_select 'status[code=ok]' }
+      end
+    end
+
+    context 'when providing the remove exclusion parameter' do
+      context 'and not having any excluded request' do
```

Comment:
What about merging this with the context in 122? We could have something like:

```
context 'when providing the remove exclusion parameter'
  context 'having any excluded request'
  context 'not having any excluded request'
```
And maybe try to share a single before block for both.

Reply:
I'll look at it

Reply:
Ah, I remember now. It will be a bit strange, if we show every context in the hierarchy we will have this:
```
context 'with an excluded request'
  context 'when providing the remove exclusion parameter'
    context 'and not having any excluded request'
```
To me its kind of strange having this excluded-but-not-excluded use case.
What do you think?

Reply:
Ok I understand it's more complex than I thought. What do you think of switching the last two contexts?

This remains the same
```
context 'with an excluded request'
  context 'when not providing the remove exclusion parameter'
  context 'when providing the remove exclusion parameter'
```

```
context 'without an excluded request'  # Before: context 'and not having any excluded request'
  context 'when providing the remove exclusion parameter' 
```

Reply:
We agreed offline to not switching the contexts